gneiss/q2/_type.py

```diff
@@ -0,0 +1,23 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, gneiss development team.
+#
+# Distributed under the terms of the GPLv3 License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+from qiime.plugin import SemanticType
+from q2_types.feature_table import (
+    FeatureTable, Frequency, BIOMV100DirFmt, BIOMV210DirFmt)
+
+
+# Regression types
+Regression = SemanticType('Regression', field_names=['type'])
+
+Linear = SemanticType('Linear',
+                      variant_of=Regression.field['type'])
+LinearMixedEffects = SemanticType('LinearMixedEffects',
+                                  variant_of=Regression.field['type'])
+
+# Tree types
+Hierarchy = SemanticType('Hierarchy', field_names=['type'])
+Cluster = SemanticType('Cluster', variant_of=Hierarchy.field['type'])
```

Comment:
Just to make sure that this is known, I'd like to point out  that you can define ad-hoc super-types with the `|` operator. e.g. `Phylogony[Rooted] | HierarchicalCluster | Graph`, etc. This might be the best way to go for now, but it doesn't help us with naming the other pieces. 

I like the idea of `Hiearchy`, but what other kinds of variant-types would we have besides `Cluster`? Ideally this could be reused for a classifier, as long as the data was strictly a hierarchy of clusters and not much else.